Adding zerork to ablate

kolosret opened the pull request:
Adding new zerorkEOS,  for more efficient chemistry solve. Adding the associated tests. Refactoring the soot process into a chemistryModel.

kolosret committed:
testing commits on clion...

kolosret committed:
testing commits on clion...

kolosret committed:
testing commits on clion...

kolosret committed:
rebase on origin/main for updated ablate

kolosret committed:
testing commits on clion...

kolosret committed:
testing commits on clion ...

v3

kolosret committed:
testing commits on clion ...

v3

kolosret committed:
testing commits on clion ...

v3

kolosret committed:
good build

kolosret committed:
good build

kolosret committed:
good build

kolosret committed:
progress

kolosret committed:
progress

kolosret committed:
progress

kolosret committed:
progress

kolosret committed:
progress

kolosret committed:
progress

kolosret committed:
progress

kolosret committed:
progress

kolosret committed:
progress

kolosret committed:
progress

kolosret committed:
progress

kolosret committed:
progress

kolosret committed:
zerork class works

kolosret committed:
updated input

kolosret committed:
zerork class works

kolosret committed:
zerork tested with 0D reactors

kolosret committed:
zerork tested with 0D reactors

kolosret committed:
zerork tested with 0D reactors

kolosret committed:
zerork tested with 0D reactors

kolosret committed:
Merge remote-tracking branch 'origin/zerork' into zerork

kolosret committed:
zerork tested with 0D reactors

kolosret committed:
zerork tested with 1D domain but not soot

kolosret committed:
zerork tested with 1D domain but not soot

kolosret committed:
zerork 1D test with soot for opt build

kolosret committed:
added extra options for zerork

kolosret committed:
added extra options for zerork -added option for load balancing in zerork

kolosret committed:
tested for mixture fraction calculator, soot process and started adding integration tests and regression tests

kolosret committed:
get rid of tchem2

kolosret committed:
tchambase fix

kolosret committed:
moving findchrestcompilerflags before cppparser so it uses my for forks the compier flags that includes -Wno-unused-variable

kolosret committed:
update to test mechanism path

kolosret committed:
fixing errors with parsin files, finishing tests

kolosret committed:
cleaning up changes 1

kolosret committed:
ablate build now includes zerork

kolosret committed:
ablate build now includes zerork

kolosret committed:
updated git commit for zerork

kolosret committed:
additional timingevent for zerork

kolosret committed:
updating zerork GIT_TAG

kolosret committed:
adding gpu option to sourcecalculator

kolosret committed:
adding gpu option to sourcecalculator

kolosret committed:
Merge remote-tracking branch 'origin/zerork' into zerork

klbudzin commented on src/eos/tChem/sourceCalculator.cpp:
Wouldn't this mean there is no default if we are just throwing an error.

kolosret replied:
there is a default, which is set, this errors out when you are trying to specify something that is not valid, instead of defaulting to stuff. I think it is better to throw an error in this case.

klbudzin commented on CMakeLists.txt:
Probably should get rid of comment and extra spacing

klbudzin commented on config/findZerork.cmake:
We don't need the final else if, and the first else if should be able to just be an else right?

klbudzin commented on src/eos/tChem.hpp:
Get rid of comment above removed code

klbudzin commented on src/eos/tChemBase.hpp:
Remove the abstracted code instead of just commenting it out

klbudzin commented on src/eos/zerork.hpp:
```diff
@@ -0,0 +1,333 @@
+
+    int nSpc;
+
+//    struct constraints {
```
git rid of unused code

klbudzin commented on src/eos/zerork.hpp:
```diff
@@ -0,0 +1,333 @@
+     */
+    [[nodiscard]] virtual const std::vector<std::string>& GetProgressVariables() const override { return ablate::utilities::VectorUtilities::Empty<std::string>; }
+
+    /**
```
out of place comment definition?

klbudzin commented on src/eos/zerork/sourceCalculatorZeroRK.cpp:
Remove unused (commented out)  or not purposeful code

klbudzin replied:
Might also be nice, to throw specific errors in setup of zerork in the constructor when something goes wrong instead of incrementing a global error state, though I expect there to be a small chance something went wrong by the time this is called

kolosret replied:
I want to keep in the code for the density renormalization, that is something I want to figure out in the future.

kolosret replied:
regarding those errors they are fake, for now..... In the future might be a good idea however for now they always return ZERORK_STATUS_SUCCESS

kolosret committed:
fixing gpu init for zerork, formating fixes

kolosret committed:
clang formated file updates

kolosret committed:
ablate.dev zerork file

kolosret committed:
oops

kolosret committed:
commit

kolosret committed:
Clang formating all the files v2

kolosret committed:
Clang formating all the files v3

kolosret committed:
apply petsc api update patch

kolosret committed:
regression test update

kolosret committed:
regression test update

kolosret committed:
integration test updates

kolosret committed:
Fixes to sourcecalulator failing

mmcgurn committed:
working on findZerork

kolosret committed:
quick fix for comminication and error handleing between reactors. Not fully tested.

kolosret committed:
changes to zerork interface in sourcecalculator

kolosret committed:
combining all build options for zerork into one cmake file

kolosret committed:
finished cmake files

kolosret committed:
finished cmake files

kolosret committed:
Merge branch 'main' into zerork

kolosret committed:
updated git tags for latest zerork

mmcgurn committed:
removed superlu root, could not run make

mmcgurn committed:
formatting

kolosret committed:
Merge pull request #3 from mmcgurn/mcgurn/zerork

zeroRk support in docker and macOS

kolosret committed:
fixing comment

kolosret committed:
updating git tags for zerork build

kolosret committed:
Adding extra options for user for CVODE integration in zerork

kolosret committed:
fix to chemtab, removing [[nodiscard]]

kolosret committed:
fix to chemistryModel.hpp pure virtual class issue

kolosret committed:
formatting fix with chemistryModel.hpp

kolosret committed:
Clarification in Gtest assert

klbudzin committed:
Soot and petsc fixes (#527)

* Adding Ability to change nucleation and surface growth pre exponential factors at run time

* Fixing Petsc API Change

* Adding in calculation of decoding steps before outputting initial file for Q.o.L.

* Formatting & Version Bump

* Fixed Issue with inertialParticles2D test that developed after petsc commit 3d0e8ed9a71e0061f1c9af54b192225135b869d5:
Which fixed not localizing coordinates for DMPlex_Copy. This causes the periodic domain to be treated as a  toroid, which does not work well with the swarm. Added a non-periodic test for the particles since they probably won't be ever used for periodic problems. Also fixed a naming issue in the particle tests.

* Turning off the periodicity tests for now and opening an issue to fix them according to most recent petsc changes.

klbudzin committed:
Klbudzin disable arm 64 (Temporarily) (#529)


The github runner machines for arm 64 went down and we are disabling these tests for now since we don't the passwords for the machines/ are unaware of which machines they were on.

* DMPlex View output has been changed to be outputted in an ordered manner, these tests were effected.

klbudzin committed:
Petsc fixes (#530)

* Petsc API changes to DMPlexCreateBoxMesh and change in typo from dim->dof in dmViewFromOptions in petsc

* The HDF5 data storage formatting in petsc was changed recently to ignore redundant fields such as cell-fields which hold our field seperated values. Until I can see how these different outputting versions can be used in ablate (i.e. smaller restart files only using aux/solution vectors) we will just default to the legacy version

* version update

kolosret committed:
Updated error handling in ZeroRK source calculator, updated default values

kolosret committed:
Version bump
